qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -148,6 +151,129 @@ def model_names(self) -> List[str]:
         """Return model names."""
         return [getattr(m, "_name", f"model-{i}") for i, m in enumerate(self._models)]
 
+    def set_options(self, **fields):
+        """Set the analysis options for :meth:`run` method.
+
+        Args:
+            fields: The fields to update the options
+
+        Raises:
+            KeyError: When removed option ``curve_fitter`` is set.
+        """
+        if fields.get("plot_residuals") and not self.options.get("plot_residuals"):
+            # checking there are no subplots for the figure to prevent collision in subplot indices.
+            if self.plotter.options.get("subplots") != (1, 1):
+                warnings.warn(
+                    "Residuals plotting is currently supported for analysis with 1 subplot.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+                fields["plot_residuals"] = False
+            else:
+                self._add_residuals_plot_config()
+        if not fields.get("plot_residuals", True) and self.options.get("plot_residuals"):
+            self._remove_residuals_plot_config()
+
+        super().set_options(**fields)
+
+    def _add_residuals_plot_config(self):
+        """Configure plotter options for residuals plot."""
+        # check we have model to fit into
+        if self.models:
+            # Cache figure options.
+            self._plot_config_cache["figure_options"] = {}
+            self._plot_config_cache["figure_options"]["ylabel"] = self.plotter.figure_options.get(
+                "ylabel"
+            )
+            self._plot_config_cache["figure_options"][
+                "series_params"
+            ] = self.plotter.figure_options["series_params"]
+            self._plot_config_cache["figure_options"]["sharey"] = self.plotter.figure_options[
+                "sharey"
+            ]
+
+            self.plotter.set_figure_options(
+                ylabel=[
+                    self.plotter.figure_options.get("ylabel", ""),
+                    "Residuals",
+                ],
+            )
+            model_names = self.model_names()
+            for model_name in model_names:
+                self.plotter.set_figure_options(
+                    sharey=False,
+                    series_params={
+                        **self.plotter.figure_options["series_params"],
+                        **{
+                            model_name: {
+                                **{
```

Comment:
What's the difference between
```python
**{
    "canvas": 0,
},
```
and
```python
"canvas": 0
```
?

Reply:
The difference  is that with `**` its the union of the fields, so if the user defined some changes to the plotter, it will affect the new plot. for example if we have a rabi experiment we want to change its plotter configuration:
```python

# Retrieve the plotter from the analysis instance
plotter = rabi.analysis.plotter

# Change the x-axis unit values
rabi.analysis.plotter.set_figure_options(
    xval_unit="arb.",
    xval_unit_scale=False   # Don't scale the unit with SI prefixes
)

# Change the color and symbol for the cosine
rabi.analysis.plotter.figure_options.series_params.update(
    {"cos": {"symbol": "x", "color": "r"}}
)

# Set figsize directly so we don't overwrite the entire style
rabi.analysis.plotter.options.style["figsize"] = (6,4)
```
to keep the curve with the color that the user specified, we need to use the union over `"series_params"` with the model name:
```python 
...
model_name: {
    **{
        "canvas": 0,
    },
     **self.plotter.figure_options["series_params"].get(model_name, {}),
}, 
...
```